src/class/hid/hid_device.c

```diff
@@ -241,8 +241,7 @@ bool hidd_control_request(uint8_t rhport, tusb_control_request_t const * p_reque
       break;
 
       case  HID_REQ_CONTROL_SET_REPORT:
-        TU_VERIFY(p_request->wLength <=sizeof(p_hid->epout_buf));
-        tud_control_xfer(rhport, p_request, p_hid->epout_buf, p_request->wLength);
+        tud_control_xfer(rhport, p_request, p_hid->epout_buf, tu_min16(sizeof(p_hid->epout_buf), p_request->wLength));

```

Comment:
What happens in this case? Does the discarded data cause misbehavior, so a NAK would be more appropriate? Does the data field have a length field inside of which would later cause issues? (This may be perfectly fine... I'm not too familiar with the HID code).

Reply:
Ah you are right, somehow  I thought it is a IN transaction :D, I need more sleep :)